apis/workloads/v1alpha1/instanceset_types.go

```diff
@@ -75,6 +75,66 @@ type SchedulingPolicy struct {
+
+const (
+	// RollingUpdateInstanceSetStrategyType indicates that update will be
+	// applied to all Pods in the InstanceSet with respect to the InstanceSet
+	// ordering constraints. When a scale operation is performed with this
+	// strategy, new Pods will be created from the specification version indicated
+	// by the InstanceSet's updateRevision.
+	RollingUpdateInstanceSetStrategyType InstanceSetUpdateStrategyType = "RollingUpdate"
+	// OnDeleteInstanceSetStrategyType triggers the legacy behavior. Version
+	// tracking and ordered rolling restarts are disabled. Pods are recreated
+	// when they are manually deleted. When a scale
+	// operation is performed with this strategy, specification version indicated
+	// by the InstanceSet's currentRevision.
+	OnDeleteInstanceSetStrategyType InstanceSetUpdateStrategyType = "OnDelete"
+)
+
+// InstanceUpdateStrategy indicates the strategy that the InstanceSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type InstanceUpdateStrategy struct {
+	// Type indicates the type of the InstanceUpdateStrategy.
+	// Default is RollingUpdate.
+	// +optional
+	Type InstanceSetUpdateStrategyType `json:"type,omitempty"`
+	// RollingUpdate is used to communicate parameters when Type is RollingUpdateStatefulSetStrategyType.
+	// +optional
+	RollingUpdate *RollingUpdateStrategy `json:"rollingUpdate,omitempty"`
+	// Members(Pods) update strategy.
+	//
+	// - serial: update Members one by one that guarantee minimum component unavailable time.
+	// - bestEffortParallel: update Members in parallel that guarantee minimum component un-writable time.
+	// - parallel: force parallel
+	//
+	// +kubebuilder:validation:Enum={Serial,BestEffortParallel,Parallel}
+	// +optional
+	MemberUpdateStrategy *MemberUpdateStrategy `json:"memberUpdateStrategy,omitempty"`
+}
+
+// RollingUpdateStrategy is used to communicate parameter for RollingUpdateInstanceSetStrategyType.
+type RollingUpdateStrategy struct {
+	// Partition indicates the ordinal at which the InstanceSet should be partitioned
+	// for updates. During a rolling update, all pods from ordinal Replicas-1 to
+	// Partition are updated. All pods from ordinal Partition-1 to 0 remain untouched.
+	// This is helpful in being able to do a canary based deployment. The default value is 0.
+	// +optional
+	Partition *int32 `json:"partition,omitempty"`
+	// The maximum number of pods that can be unavailable during the update.
+	// Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
+	// Absolute number is calculated from percentage by rounding up. This can not be 0.
+	// Defaults to 1. The field applies to all pods in the range 0 to Replicas-1.
+	// That means if there is any unavailable pod in the range 0 to Replicas-1,
+	// it will be counted towards MaxUnavailable.
+	// +optional
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
```

Comment:
always involved restarting pods, and any update to a pod would cause it to become unavailable. Therefore, controlling the number of maxUnavailable pods represented the level of concurrency.

However, in InstanceSet, if we continue to use 'maxUnavailable', strictly speaking, all non-restarting updates (i.e., those that don't cause the pod to become unavailable) would not be controlled by this parameter.

Reply:
Why should we control the `non-restarting updates` as they are taking effect instantly after the Pods being updated? Do all the `non-restarting updates` in one reconciliation loop seems no difference with doing them in several reconciliation loops.

Reply:
Because 'non-restarting update' is also a change in production environment, it should be controlled by a gradual upgrade strategy. For example, it's reasonable to allow users to make changes to a small number of replicas first, and after verification, gradually roll out to more replicas.

e.g. A 'non-restarting update' might involve parameter modifications, which could potentially lead to issues such as performance degradation.

Reply:
To summarize, incorrect 'non-restarting updates' can potentially harm business continuity. Therefore, users may require a gradual update process. This is something we need to consider.

Reply:
@weicao I think, for the 'non-restarting update,' it seems that we don't have a way to automate the control. How about we use `partition` to manually control it instead?

Reply:
> @weicao I think, for the 'non-restarting update,' it seems that we don't have a way to automate the control. How about we use `partition` to manually control it instead?

When you have an instance template, how do you plan to use partitions?

Reply:
What I mean is, say if you have 3 templates and a total of 10 replicas, when you set the partition to 7 (upgrading the 8th, 9th, and 10th replicas), it becomes difficult to determine which templates these 8th, 9th, and 10th replicas belong to. This makes it challenging to verify whether the updated replicas meet the expected outcomes.

Reply:
> What I mean is, say if you have 3 templates and a total of 10 replicas, when you set the partition to 7 (upgrading the 8th, 9th, and 10th replicas), it becomes difficult to determine which templates these 8th, 9th, and 10th replicas belong to. This makes it challenging to verify whether the updated replicas meet the expected outcomes.

@weicao We have also considered this issue. However, based on our current requirements, we do not need to specify a separate partition for gray upgrades in multiple templates. Nevertheless, I understand the need for multi-template partitioning, and we can discuss and design a solution for this.

Reply:
Currently, our requirement is only for the partition to be able to perform a global update in alphabetical order.